python/cudf/cudf/core/internals/where.py

```diff
@@ -0,0 +1,375 @@
+        )
+        if len(common_cols) > 0:
+            # If `frame` and `cond` are having unequal index,
+            # then re-index `cond`.
+            if not frame.index.equals(cond.index):
+                cond = cond.reindex(frame.index)
+        else:
+            if cond.shape != frame.shape:
+                raise ValueError(
+                    """Array conditional must be same shape as self"""
+                )
+            # Setting `frame` column names to `cond`
+            # as `cond` has no column names.
+            cond.columns = frame.columns
+
+        (source_df, others,) = _normalize_columns_and_scalars_type(
+            frame, other
+        )
+        if isinstance(other, cudf.core.frame.Frame):
+            others = others._data.columns
+
+        out_df = cudf.DataFrame(index=frame.index)
+        if len(frame._columns) != len(others):
+            raise ValueError(
+                """Replacement list length or number of dataframe columns
+                should be equal to Number of columns of dataframe"""
+            )
+        for i, column_name in enumerate(frame._data.names):
+            input_col = source_df._data[column_name]
+            other_column = others[i]
+            if column_name in cond._data:
+                if isinstance(input_col, cudf.core.column.CategoricalColumn):
+                    if cudf.utils.dtypes.is_scalar(other_column):
+                        try:
+                            other_column = input_col._encode(other_column)
+                        except ValueError:
+                            # When other is not present in categories,
+                            # fill with Null.
+                            other_column = None
+                        other_column = cudf.Scalar(
+                            other_column, dtype=input_col.codes.dtype
+                        )
+                    elif hasattr(other_column, "codes"):
+                        other_column = other_column.codes
+                    input_col = input_col.codes
+
+                result = cudf._lib.copying.copy_if_else(
+                    input_col, other_column, cond._data[column_name]
+                )
+
+                if isinstance(
+                    frame._data[column_name],
+                    cudf.core.column.CategoricalColumn,
+                ):
+                    result = cudf.core.column.build_categorical_column(
+                        categories=frame._data[column_name].categories,
+                        codes=cudf.core.column.as_column(
+                            result.base_data, dtype=result.dtype
+                        ),
+                        mask=result.base_mask,
+                        size=result.size,
+                        offset=result.offset,
+                        ordered=frame._data[column_name].ordered,
+                    )
+            else:
+                from cudf._lib.null_mask import MaskState, create_null_mask
```

Comment:
Can we move this import out of the function? Either by using top level imports or by using absolute paths to these functions as needed.